File: crates/domain/src/inventory/warehouse.rs
use crate::shared::ids::WarehouseId;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Warehouse {
    pub id: WarehouseId,
    pub name: String,
    pub address: Option<String>,
    pub is_active: bool,
    pub is_default: bool,
}

File: packages/shared-types/src/warehouse.ts
export interface WarehouseDto {
  id: string;
  name: string;
  address?: string | null;
  is_active: boolean;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateWarehouseRequest {
  name: string;
  address?: string | null;
}

export interface UpdateWarehouseRequest {
  id: string;
  name: string;
  address?: string | null;
  is_active: boolean;
  is_default: boolean;
}
